front/poke/temporal/activities.ts

```diff
@@ -40,3 +85,424 @@ export async function scrubDataSourceActivity({
+                    }
+                    await AgentRetrievalAction.destroy({
+                      where: { id: agentMessage.agentRetrievalActionId },
+                      transaction: t,
+                    });
+                  }
+                  await agentMessage.destroy({ transaction: t });
+                }
+              }
+              if (msg.contentFragmentId) {
+                await ContentFragment.destroy({
+                  where: { id: msg.contentFragmentId },
+                  transaction: t,
+                });
+              }
+              await MessageReaction.destroy({
+                where: { messageId: msg.id },
+                transaction: t,
+              });
+              await Mention.destroy({
+                where: { messageId: msg.id },
+                transaction: t,
+              });
+              await msg.destroy({ transaction: t });
+            }
+            await ConversationParticipant.destroy({
+              where: { conversationId: c.id },
+              transaction: t,
+            });
+            await c.destroy({ transaction: t });
+          })();
+        })
+      );
+    }
+  });
+}
+
+export async function deleteAgentsActivity({
+  workspaceId,
+}: {
+  workspaceId: string;
+}) {
+  const auth = await Authenticator.internalBuilderForWorkspace(workspaceId);
+  const workspace = auth.workspace();
+
+  if (!workspace) {
+    throw new Error("Could not find the workspace.");
+  }
+
+  const agents = await AgentConfiguration.findAll({
+    where: {
+      workspaceId: workspace.id,
+    },
+  });
+  await front_sequelize.transaction(async (t) => {
+    for (const agent of agents) {
+      if (agent.generationConfigurationId) {
+        await AgentGenerationConfiguration.destroy({
+          where: {
+            id: agent.generationConfigurationId,
+          },
+          transaction: t,
+        });
+      } else if (agent.retrievalConfigurationId) {
+        await AgentDataSourceConfiguration.destroy({
+          where: {
+            retrievalConfigurationId: agent.retrievalConfigurationId,
+          },
+          transaction: t,
+        });
+        await AgentRetrievalConfiguration.destroy({
+          where: {
+            id: agent.retrievalConfigurationId,
+          },
+          transaction: t,
+        });
+      } else if (agent.dustAppRunConfigurationId) {
+        await AgentDustAppRunAction.destroy({
+          where: {
+            dustAppRunConfigurationId: agent.dustAppRunConfigurationId,
+          },
+          transaction: t,
+        });
+        await AgentDustAppRunConfiguration.destroy({
+          where: {
+            id: agent.dustAppRunConfigurationId,
+          },
+          transaction: t,
+        });
+      }
+      await AgentUserRelation.destroy({
+        where: {
+          agentConfigurationId: agent.id,
+        },
+        transaction: t,
+      });
+      await agent.destroy({ transaction: t });
```

Comment:
I think we want to delete the GlobalAgentSettings before we delete the agent otherwise it will refuse the deletion no?

Reply:
Would also add a log line here?

Reply:
Global agents settings are only related to global agents not to the agents in db so we should be good but I'll move it before!